build.gradle

```diff
@@ -193,6 +198,17 @@ task fatJar(type: Jar) {
     with jar
 }
 
+shadowJar {
```

Comment:
Could you explain why this is needed?  I assume it's to prevent conflicts with hadoop dependencies in some way, but I don't really know what it's doing exactly.